### PR TITLE
Automate publishing snaps to the stable channel

### PR DESCRIPTION
**Pull request by bmw**
Fixes https://github.com/certbot/certbot/issues/8171.

Unfortunately the diff between the new and old version of the script is so large that the diff isn't tracking changes across the file rename. An easier way to review this PR might be:

1. Notice that the first commit in this PR https://github.com/certbot/certbot/pull/8351/commits/e85ee0e2e9e4f13c6d31d501b70b53d9ac0c0f72 renames the script with no other changes.
2. Look at https://github.com/certbot/certbot/compare/e85ee0e2e9e4f13c6d31d501b70b53d9ac0c0f72...snap-automation to see the diff between that commit and the rest of this branch.

See the comment at the top of the script to learn how to set things up and run this. Running the script between releases will have no effect on our snaps and it should fail when creating the GitHub release. The latter is described at https://github.com/certbot/certbot/pull/8189#discussion_r466707114.

Once this PR is merged, I will update our release instructions to use this script.

To avoid making unexpected changes during testing, this script just publishes the Certbot snap to the stable channel for now, but after the release tomorrow we can revert https://github.com/certbot/certbot/pull/8351/commits/6422728fedb76244fb545dfd7308d782451e9a18 if we want and run the script to publish the DNS plugin snaps to the stable channel as well. I personally think we may as well do this. See https://github.com/certbot/certbot/issues/8283.

@ohemorange, since you and I will be the ones using this script, I think it makes the most sense for you to review this when you find the time.

### CODE REVIEW

File: tools/finish_release.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    :rtype: `list` of `str`
+
+    :raises subprocess.CalledProcessError: if the snapcraft command
+        fails
+
+    :raises AssertionError: if the expected snaps are not found
+
+    """
+    cmd = ['snapcraft', 'status', snap]
+    process = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, universal_newlines=True)
```
Comment (ohemorange) [inline tools/finish_release.py:144]: I know it should be covered by running `assert_logged_into_snapcraft` first, but just as a double check, could we print something like "getting snap revisions" here so we know where we've stalled?

**Review by ohemorange — CHANGES_REQUESTED** [review]
One small suggestion, and also we can revert that certbot-only commit you mentioned, but otherwise looks good!

File: tools/finish_release.py (hunk 1 above)
Comment (bmw) [reply tools/finish_release.py:143]: Good idea! I did this.

**Comment by bmw** [comment]
Thanks for the review!

Just to be clear, are you OK with us moving our DNS plugins for the last release to the stable channel? If so, I will go ahead and do that. You're also welcome to do it yourself if you want.

After this is done, I'll create issues to track removing `--beta` from our wildcard instructions on the website.

**Comment by ohemorange** [comment]
> Just to be clear, are you OK with us moving our DNS plugins for the last release to the stable channel? If so, I will go ahead and do that. You're also welcome to do it yourself if you want.

Yeah, I actually forgot until doing the release that we hadn't already done that. I think they're ready!

**Review by ohemorange — APPROVED** [review]

**Pull request merged**